groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -336,12 +422,20 @@ typename add_pointer<const TYPE>::type get_if(
 template <class HEAD, class... TAIL>
 bool operator==(const variant<HEAD, TAIL...>& lhs,
                 const variant<HEAD, TAIL...>& rhs);
+    // Two variant objects compare equal if they are both
+    // 'valueless_by_exception', or if they have the same active alternative
+    // and the alternative objects compare equal.  Otherwise, two variant
+    // objects do not compare equal.
+    // All alternatives shall support `operator==`.
+
```

Comment:
Following the above principles, this docstring might look something like this:
```suggestion
    // Return 'true' if the specified 'lhs' and 'rhs' are both
    // 'valueless_by_exception' or if they have the same active alternative
    // and the alternative objects compare equal; otherwise, return 'false'.
    // All alternatives shall support `operator==`.

```
ditto for the other operators' docstrings.

I'll refrain from pointing out the "specified" thing for the rest of this review  - any stragglers can be fixed when we merge upstream.